javascript/MaterialXView/source/viewer.js

```diff
@@ -158,17 +161,65 @@ export class Scene
         });
     }
 
-    // For now every mesh uses the same material
-    updateMaterial(material)
+    // Determine string DAG path based on individual node names.
+    getDagPath(node)
+    {
+        let path = [node.name];
+        while (node.parent)
+        {
+            node = node.parent;
+            if (node)
+            {
+                path.unshift(node.name + DAG_PATH_SEPERATOR);
+            }
+        }
+        return path;
+    }
+
+    // Assign material shader to associated geometry
+    updateMaterial(matassign)
     {
+        let assigned = 0;
+
+        const shader = matassign.getShader();
+        const material = matassign.getMaterial().getName();
+        const geometry = matassign.getGeometry();
+
         const scene = this.getScene();
         const camera = this.getCamera();
         scene.traverse((child) => {
-            if (child.isMesh) {
-                child.material = material;
-                child.material.needsUpdate = true;
+            if (child.isMesh) 
+            {
+                const dagPath = this.getDagPath(child);
```

Comment:
I will keep dag path support a separate update for now as first the glTF files needs to be fixed to have pathing which match the materialassign notation and hierarchy, and the glTF loader for desktop needs to also set proper dag path names.